endpoints/10-cluster-endpoints.md

```diff
@@ -0,0 +1,319 @@
+    "action" : "apply",
+    "flags" : ""
+  }
+]
+```
+
+
+**Reply:**  Long indicating the ProActive Job Id (Generated workflow in Proactive)
+
+Field `appName` must be valid as a filename; therefore, spaces, quotes, and other special characters should not be used.
+
+`action` value `apply` can be used both for the initial deployment and for ongoing application management. For example, you can adjust the number of replicas to scale the application according to demand.
+
+Note that from the `appFile` the value of `{{app_component_name}}` is used when calling LabelNode endpoint. Also, it is to include `\n` characters at the end of each line to indicate line breaks (JSON requirement).
+
+#### 2.2- GetAllClouds endpoint:
+
+### 10.5- DeleteCluster endpoint:
+
+**Description**: This endpoint allows users to delete an existing Kubernetes cluster deployment. It removes all resources associated with the specified cluster, including nodes, network configurations, and any deployed applications. By using this endpoint, users can clean up and release the resources used by the cluster.
+
+
+
+**Path:**
+
+```url
+🔴 DEL {{protocol}}://{{sal_host}}:{{sal_port}}/sal/cluster/{{cluster_name}}
+```
+
+**Headers:** `sessionid`
+
+**Body:** None
+
+
+**Reply:**  Boolean, `true` if successful
+
+
+### 10.6- ScaleOut endpoint:
+
+**Description**: This endpoint allows users to dynamically expand their Kubernetes cluster by adding new worker nodes.
+This scaling operation is based on existing worker node definitions and is critical when increasing the cluster's capacity to support more replicas for applications.
+Each worker node is introduced using the existing worker node candidate, but is uniquely identified by its `nodeName`. The number of nodes added corresponds to the number of replicas the user wishes to create.
+For example, if a user wants to increase the number of replicas from 1 to 3, they could call the ScaleOut endpoint to add two more worker nodes (e.g., `worker2`, `worker3`), ensuring the cluster can support the desired number of replicas.
+
+After the ScaleOut operation, the new worker nodes and their status can be tracked using the GetCluster endpoint or monitored in the ProActive. The created workflows correspond to the one which is used to initial node deployments in cluster, using user defined scripts.
+When the new nodes are deployed, the LabelNodes endpoint should be used to label these nodes (e.g., `worker2_name`, `worker3_name`) for identification and management purposes.
+To complete ScaleOut process, ManageApplication endpoint is to be called, with updated number of the replicas (e.g. by adding two new replicas).
+
+
+**Path:**
+
+```url
+🟡 POST {{protocol}}://{{sal_host}}:{{sal_port}}/sal/cluster/{{cluster_name}}/scaleout
+```
+
+**Headers:** `sessionid`
+
+**Body:** Json input following this format:
+
+```json
+[
+  //scale out using WorkerNodeCandidate which is part of the cluster and introducing the new node name
```

Comment:
json does not allow comments (similar remark as below).

You can do instead for example:
```

**Body:** Json input following this format:

Note: scale out using WorkerNodeCandidate which is part of the cluster and introducing the new node name

```json
[
...


```